src/Castle.Core/DynamicProxy/Generators/AttributesToAvoidReplicating.cs

```diff
@@ -18,10 +18,11 @@ namespace Castle.DynamicProxy.Generators
 	using System.Collections.Generic;
 	using System.Linq;
 	using System.Reflection;
-	using System.Threading;
 
 	public static class AttributesToAvoidReplicating
 	{
+		private static readonly object lockObject = new object();
```

Comment:
Is this needed? I think you could just lock on `attributes` directly.

Reply:
As the list is not accessible externally of the class this is of course a viable alternative.

The only reason not to do so is to prevent future issues where collections could be made available externally and an external consumer could lock it.

I'm fine either way. I won't go around locking other "people's" objects, so I most likely won't be dealing with that issue, should it ever arise ;-)